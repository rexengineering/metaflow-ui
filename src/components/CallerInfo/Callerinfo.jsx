import React, {useEffect, useState} from "react";
import PropTypes from "prop-types";
import { Box, Divider, makeStyles, Typography } from "@material-ui/core";
import Task from "../Task";
import WorkflowInstantiator from "../WorkflowInstantiator";
import Workflow from "../Workflow";
import {initWorkflow} from "../../store/thunks/thunks";
import getDeploymentId from "../../store/thunks/getDeploymentId";
import {useDispatch, useSelector} from "react-redux";
import {calculateWorkFlowNameFromDeploymentID} from "../../utils/tasks";
import {selectWorkflowID} from "../../store/selectors/rexflow";

const useStyles = makeStyles((theme) => ({
  container: {
    backgroundColor: theme.palette.grey["800"],
    padding: theme.spacing(7, 2, 2),
    height: "100%",
  },
  title: {
    color: theme.palette.grey["300"],
    margin: theme.spacing(4, 0),
  },
  subtitle: {
    color: theme.palette.grey["400"],
  },
  divider: {
    backgroundColor: theme.palette.grey["400"],
  },
}));

function CallerInfo({ callerName, deploymentID }) {
  const dispatch = useDispatch();
  const classes = useStyles();
  const [workflowName, setWorkflowName] = useState(null);
  const workflowID = useSelector(selectWorkflowID(workflowName));

  useEffect(() => {
      if (!deploymentID) return;
      setWorkflowName(calculateWorkFlowNameFromDeploymentID(deploymentID));
  }, [deploymentID, setWorkflowName]);

  return (
    <Box className={classes.container}>
      <Typography className={classes.subtitle} variant="body2">
        You are speaking with:
      </Typography>
      <Typography className={classes.title} variant="h3">
        {callerName}
      </Typography>
      <Divider className={classes.divider} />
      <Workflow workflowID={workflowID} />
    </Box>
  );
}

CallerInfo.propTypes = {
  callerName: PropTypes.string.isRequired,
  deploymentID: PropTypes.string.isRequired,
};

export default CallerInfo;
