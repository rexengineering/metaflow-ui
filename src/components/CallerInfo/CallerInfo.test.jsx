import { render, screen } from "@testing-library/react";
import React from "react";
import CallerInfo from "./Callerinfo";
import {Provider} from "react-redux";
import getStore from "../../store";


describe("<CallerInfo />", () => {

    const renderCallerInfo = (callerName = "John Doe", deploymentID = "deployment-865") => (
        <Provider store={getStore({})}>
            <CallerInfo callerName={callerName} workflowName={deploymentID}/>
        </Provider>
    );

    it("should render", () => {
        const { asFragment } = render(renderCallerInfo());
        const initialRender = asFragment();

        expect(initialRender).not.toBeNull();
    });

    it("should match snapshot", () => {
        const { asFragment } = render(renderCallerInfo());
        const initialRender = asFragment();

        expect(initialRender).toMatchSnapshot();
    });

    it("should display caller name properly", () => {
        const callerName = "Gabriel";
        render(renderCallerInfo(callerName));
        expect(screen.getByText(callerName)).toBeInTheDocument();
    });
});
