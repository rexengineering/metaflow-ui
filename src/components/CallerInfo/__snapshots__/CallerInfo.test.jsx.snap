// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CallerInfo /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-15 makeStyles-container-11"
  >
    <p
      class="MuiTypography-root makeStyles-subtitle-13 MuiTypography-body2"
    >
      You are speaking with:
    </p>
    <h3
      class="MuiTypography-root makeStyles-title-12 MuiTypography-h3"
    >
      John Doe
    </h3>
    <hr
      class="MuiDivider-root makeStyles-divider-14"
    />
    <section>
      <h6
        class="MuiTypography-root MuiTypography-h6"
      />
      <section
        class="makeStyles-inProgressContainer-19"
      >
        <div
          class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
          role="progressbar"
          style="width: 25px; height: 25px;"
        >
          <svg
            class="MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
              cx="44"
              cy="44"
              fill="none"
              r="20.2"
              stroke-width="3.6"
            />
          </svg>
        </div>
        <p
          class="MuiTypography-root makeStyles-inProgressMessage-20 MuiTypography-body2"
        >
          Workflow in progress
        </p>
      </section>
    </section>
  </div>
</DocumentFragment>
`;
